plugins:
  - Movement
#  - Elevator
#  - Poi
#  - Ptz

pad:
  type: ps4
  num_of_buttons: 14
  num_of_axes: 14
  joy_topic: joy
  joy_timeout: 5.0


Movement:
  type: robotnik_pad_plugins/Movement
  max_linear_speed: 1.5
  max_angular_speed: 3
  cmd_topic_vel: pad_teleop/cmd_vel
  config:
    button_deadman: 5
    axis_linear_x: 1
    axis_linear_y: 0
    axis_angular_z: 2
    button_speed_up: 3
    button_speed_down: 1
    button_kinematic_mode: 7
    use_accel_watchdog: true
    axis_watchdog: 8
    watchdog_duration: 0.5

Elevator:
  type: robotnik_pad_plugins/Elevator
  elevator_service_name: robotnik_base_control/set_elevator
  config:
    deadman: 5
    axis_elevator: 10

Poi:
  type: robotnik_pad_plugins/Poi
  poi_service_name: poi_interactive_marker/save_robot_pose
  config:
    deadman: 5
    save_poi_l3: 10
    save_poi_r3: 11

Ptz:
  type: robotnik_pad_plugins/Ptz
  cmd_topic_ptz: ptz/command
  
  speed_increment: 0.5
  speed_limit: 3.14

  home_pan_position: 0.0
  home_tilt_position: 0.0
  home_zoom_position: 0.0
  
  position_increment: 0.1
  position_increment_limit: 0.5
  min_pan_position: -3.14
  max_pan_position: 3.14
  min_tilt_position: -1.57
  max_tilt_position: 1.57
  max_zoom_position: 12000

  set_position_mode: false # if false, set speed mode

  config:
    button_deadman: 5
    button_home: 12
    button_pan:  10
    button_tilt: 9
    button_zoom_in: 4 
    button_zoom_out: 6
    button_increment_up: 2 
    button_increment_down: 0
    #button_ptz_mode: 8